Improve logseq tag handling

ctem opened the pull request:
This PR includes two fixes:

- Remove the `#` tag prefix to avoid "Malformed entry" errors in Taskwarrior when applying coefficients to tags.
- Use non-greedy regex to ensure page link-style tags are extracted separately.

The relevant documentation is also updated.

ryneeverett reviewed (approved):
I don't entirely follow the non-greedy regex change because I'm not familiar enough with logseq to know what "page link-style tags" are. If you added such a tag to the tests somewhere that would be a nice touch.

ryneeverett commented:
Pinging @scross01 for possible input.

ctem committed:
Remove `#` prefix from logseq tags

The prefix causes taskwarrior to throw a "Malformed entry" error when
applying a coefficient to the tag.

ctem committed:
Perform logseq page link tag search non-greedily

This ensures that page link-style tags are not inadvertently combined
into a single match.

ctem committed:
Update Tags section of logseq docs

- The leading '#' is removed from the example tags.
- Typos are corrected.

ctem committed:
Add tags to logseq test

Three logseq-compatible tags are added to the test record: two in the
'page link' format and one in the conventional 'tag' format. The
expected results demonstrate:

- issue title: page link double brackets are replaced with special
  characters for taskwarrior compatibility
- tags:
  - the conventional tag is added first
  - non-greedy regex matching prevents the inadvertent combination of
    the two page link tags into a single tag
  - the leading `#` is stripped from all tags

ctem committed:
Refactor logseq test to use dynamic URI

For improved maintainability, the logseq block URI in the expected
results is now dynamically assigned by concatenating the base URI with
the test record UUID.

ctem commented:
> "page link-style tags"

Logseq supports several syntaxes, including:

- `#PageTitle`: conventional, no-whitespace tag (called "single-word tag" in bugwarrior docs)
- `[[Page Title]]`: page link—_not treated as a tag in bugwarrior_
- `#[[Page Title]]`: "page link-style" tag for whitespace support (called "multi-word tag" in bugwarrior docs)

> the non-greedy regex change

I'll try to illustrate the issue with an example procedure:

1. Consume the logseq issue title:
   `TODO Do something #[[Test tag]] #[[TestTag]] #TestTag`
1. Format the title, replacing brackets with respective special characters:
   `TODO Do something #【Test tag】 #【TestTag】 #TestTag`
1. Extract tags from the formatted title, starting with conventional tags, then greedily matching page link-style tags (which will match from the first `#` to the last `】`, resulting in a single match that includes both tags):
   `['#TestTag', '#【Test tag】 #【TestTag】']`
1. Remove whitespace and the special characters from all tags:
   `['#TestTag', '#Testtag#TestTag']`
1. Strip the leading `#` from all tags for the final result:
   `['TestTag', 'Testtag#TestTag']`

By switching to non-greedy matching, the two page link-style tags are matched separately, resulting in the correct final list, `['TestTag', 'Testtag', 'TestTag']`, which taskwarrior will ultimately deduplicate to `['TestTag', 'Testtag']`.

Hope that makes sense!

ryneeverett reviewed (approved):
Thanks!

scross01 commented:
Nice fix. Thanks
